synfig-studio/src/gui/widgets/widget_distance.cpp

```diff
@@ -99,6 +100,27 @@ Widget_Distance::on_key_release_event(GdkEventKey* event)
 	SYNFIG_EXCEPTION_GUARD_END_BOOL(true)
 }
 
+void
+Widget_Distance::after_event(GdkEvent *event)
```

Comment:
1) As you can see, `Widget_Distance`, `Widget_Entry`, `Widget_Spin` have almost the same code, so it looks like bad practice.
2) Why do you select not all the text, but only up to the first digit?

Reply:
Regarding 2, all the digits are selected, it is the units which are not selected.

Regarding 1, well we need to connect to same signals in spin, entry, and distance widgets. The only other possible alternative I can think of is to connect instances each time (which would be really bad of course ).

Reply:
> Regarding 2, all the digits are selected


https://user-images.githubusercontent.com/5604544/194246772-e8bfc7cb-e9a7-4029-8441-5d96acec3ce9.mp4

> Should the units be selected as well ?

I think yes. Units can be selected too.

Reply:
Ok then I"ll just make it select all.

P.s. I thought it did select all digits I remember it working when I tried but maybe the zeros after the digits were an unhandled case anyways, that's not needed now if all should be selected :).

Reply:
alright, @ice0 now it selects all. so point one should be done :)

Reply:
@mohamedAdhamc It'd have been amazing if "select all" is enabled in all text fields by default.

https://user-images.githubusercontent.com/50204208/211195419-a5cd88fc-e39e-43f9-b47b-19f6b04d6c69.mp4